### PR TITLE
Cleanup build

### PR DESCRIPTION
**Pull request by jzohrab**
Clean up the installation and build:
- ignore all generated files
- add autogen.sh

### CODE REVIEW

**Comment by jzohrab** [comment]
Note: am not an expert in this, but I don't feel that auto-generated files should be committed to a repo.  Ref http://stackoverflow.com/questions/3290908/which-files-generated-by-autotools-should-i-keep-in-version-control-repository

File: INSTALL [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
+    apt-get install -y autotools
+
+Build and Install
+-----------------
+
+    $ unzip logkeys-master.zip     # or use git clone
+    $ cd logkeys-master
+    $ ./autogen.sh
+    $ ./configure
+    $ cd build         # keeps the root and src dirs clean
```
Comment (kernc) [inline INSTALL]: Shouldn't this `cd` come before `../configure`?

**Comment by kernc** [comment]
I generally agree.

File: INSTALL (hunk 1 above)
Comment (jzohrab) [inline INSTALL]: I thought I'd checked that ...  I believe it should, actually.  Will try both ways and see what happens.

**Comment by jzohrab** [comment]
I've seen both ways - I think for distributing a .tar.gz, you can have both of them.  Let's stick with this for now until someone comes with a valid complaint.

**Comment by jzohrab** [comment]
Will post here with a `ship it` or similar comment when I think it's good to go.

File: INSTALL (hunk 1 above)
Comment (kernc) [inline INSTALL]: Just noticed that the echo help in _autogen.sh_ says the other way around.

Comment (jzohrab) [inline INSTALL]: autogen usage needs to be worked out as well as it generates into root currently.  :-1:

Comment (jzohrab) [inline INSTALL]: Yeah thanks.  Must say that this is a bit embarrassing.  :-)  Been a while.  But will continue to slog through.

Comment (jzohrab) [inline INSTALL]: Fixed and rebased branched, force pushed.

**Comment by jzohrab** [comment]
@kernc this seems ok now.  Builds successfully from a completely clean project and `build` dir with no excess noise in git history.  Some of the things in `configure.ac` need to be cleaned up, eg the `AC_INIT` line, but can handle that later.

**Comment by kernc** [comment]
Thanks for this!

**Comment by kernc** [comment]
Hmm, what about _depcomp_, _install-sh_, and _missing_ files? Are those needed?

**Comment by jzohrab** [comment]
I didn't remove these files, but that doesn't mean you're wrong.  The `autogen.sh` creates these files with the line `automake --add-missing`, but these aren't committed (autogen.sh should be run in the /build dir).  Not committing may or may not be best practice, but I don't have the real knowledge to decide on this, and don't have the time to investigate in depth.  Maybe we should specify in the README that we have questions on the project overall, such as what we can do for cross-platform builds, and request that people with skills open PRs.

**Comment by kernc** [comment]
It isn't called "autohell" for no reason. :smiley:

**Pull request merged**